lib/src/modules/task.cc

```diff
@@ -128,11 +152,146 @@ static TaskCommand getTaskCommand(const fs::path &task_executable)
     return TaskCommand { task_executable.string(), { } };
 }
 
+// Converts boost filesystem errors to a task_error object.
+static Module::ProcessingError toModuleProcessingError(const fs::filesystem_error& e) {
+    auto err_code = e.code();
+    if (err_code == boost_error::no_such_file_or_directory) {
+        return Module::ProcessingError(lth_loc::format("No such file or directory: {1}", e.path1()));
+    } else if (err_code ==  boost_error::file_exists) {
+        return Module::ProcessingError(lth_loc::format("A file matching the name of the provided sha already exists"));
+    } else {
+        return Module::ProcessingError(e.what());
+    }
+}
+
+// Take out common permissions.
+static fs::perms TASK_FILE_PERMS = fs::owner_read | fs::owner_write | fs::group_read;
```

Comment:
Do you want me to go ahead and add these for Linux and ignore them for Windows? I was waiting on #623 so that I could reuse the stuff that's there, e.g. the permissions defined in configuration.hpp.